File: python/leetcode/n-ary-tree-postorder-traversal.py
"""
# Definition for a Node.
class Node(object):
    def __init__(self, val, children):
        self.val = val
        self.children = children
"""
class Solution(object):
    def postorder(self, root):
        """
        :type root: Node
        :rtype: List[int]
        """
        return self.traverse(root, [])

    def traverse(self, root, result):
        if root is not None:
            for i in range(len(root.children)):
                result = self.traverse(root.children[i], result)
            result.append(root.val)
        return result

File: java/leetcode/n-ary-tree-postorder-traversal.java
/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val,List<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/
class Solution {
    public List<Integer> postorder(Node root) {
        return traverse(root, new ArrayList<Integer>());
    }
    
    public List<Integer> traverse(Node root, List<Integer> result) {
        if (root != null) {
            for (int i = 0; i < root.children.size(); i++) {
                result = traverse(root.children.get(i), result);
            }
            result.add(root.val);
        }
        return result;
    }
}
